app/frontend/Shared/MainLayout.razor

```diff
@@ -41,6 +41,10 @@
             <MudIconButton Icon="@Icons.Custom.Brands.GitHub" Color="Color.Inherit" Size="Size.Large"
                         Title="Visit the Azure Samples: GitHub repository for this app."
                         Href="https://github.com/Azure-Samples/azure-search-openai-demo-csharp" Target="_blank" />
+            @if (ShowLogoutButton)
```

Comment:
This logic for conditional rendering should instead live within the `LogoutDisplay` itself. That would simplify the `MainLayout` as it would just call `<LogoutDisplay />`.